modules/core/src/test/scala/org/tessellation/infrastructure/trust/storage/TrustStorageSuite.scala

```diff
@@ -0,0 +1,361 @@
+    }
+  }
+
+  test(
+    "current ordinal trust storage doesn't update from next ordinal trust storage because the delay expected ordinal has not been reached"
+  ) {
+    val minOrdinalValue = snapshotInterval.value
+    val maxOrdinalValue = snapshotInterval + delayToUpdateCurrent - 1L
+
+    val gen = for {
+      trustInfo <- Gen.mapOfN(4, genPeerTrustInfo)
+      delta <- Gen.chooseNum(minOrdinalValue, maxOrdinalValue)
+      ordinal = Refined.unsafeApply[Long, NonNegative](delta)
+    } yield (TrustMap.empty.copy(trust = trustInfo), SnapshotOrdinal(ordinal))
+
+    forall(gen) {
+      case (trust, ordinal) =>
+        for {
+          store <- TrustStorage.make(trust)
+
+          _ <- store.updateNext(ordinal)
+
+          next <- store.getNextOrdinalTrust
+
+          _ <- store.updateCurrent(ordinal)
+
+          expectedCurrent <- store.getCurrentOrdinalTrust
+        } yield
+          expect.all(
+            SnapshotOrdinal(snapshotInterval |+| delayToUpdateCurrent) > ordinal,
+            trust == next.trust,
+            OrdinalTrustMap.empty == expectedCurrent
+          )
+    }
+  }
+
+  test(
+    "current ordinal trust storage updates from next ordinal trust storage because the delay expected ordinal has been reached"
+  ) {
+    val minFirstOrdinalValue = 0L
+    val maxFirstOrdinalValue = snapshotInterval - 1L
+    val minSecondOrdinalValue = snapshotInterval.value + delayToUpdateCurrent
+    val maxSecondOrdinalValue = (snapshotInterval * 2) - 1L
+
+    val gen = for {
+      trustInfo <- Gen.mapOfN(4, genPeerTrustInfo)
+
+      firstDelta <- Gen.chooseNum(minFirstOrdinalValue, maxFirstOrdinalValue)
+      firstOrdinal = Refined.unsafeApply[Long, NonNegative](firstDelta)
+
+      secondDelta <- Gen.chooseNum(minSecondOrdinalValue, maxSecondOrdinalValue)
+      secondOrdinal = Refined.unsafeApply[Long, NonNegative](secondDelta)
+    } yield (TrustMap.empty.copy(trust = trustInfo), SnapshotOrdinal(firstOrdinal), SnapshotOrdinal(secondOrdinal))
+
+    forall(gen) {
+      case (trust, firstOrdinal, secondOrdinal) =>
+        for {
+          store <- TrustStorage.make(trust)
+
+          _ <- store.updateNext(firstOrdinal)
+
+          firstNext <- store.getNextOrdinalTrust
+
+          _ <- store.updateCurrent(secondOrdinal)
+
+          current <- store.getCurrentOrdinalTrust
+          secondNext <- store.getNextOrdinalTrust
+        } yield
+          expect.all(
+            SnapshotOrdinal(snapshotInterval |+| delayToUpdateCurrent) <= secondOrdinal,
+            firstNext.trust == current.trust,
+            SnapshotOrdinal(snapshotInterval) == current.ordinal,
+            PublicTrustMap.empty == current.peerLabels,
+            SnapshotOrdinal(Refined.unsafeApply[Long, NonNegative](snapshotInterval * 2)) == secondNext.ordinal,
+            TrustMap.empty == secondNext.trust
+          )
+    }
+  }
+
+  test(
+    "next ordinal trust storage doesn't update from gossip because the gossip ordinal doesn't match"
```

Comment:
`because the gossip ordinal doesn't match` or `when the gossip ordinal doesn't match`?

Reply:
Changed the test name to `next ordinal trust storage doesn't update from gossip when the gossip ordinal doesn't match the next ordinal`.